Fix/#2116 xss vulnerabilities in 2 8

Limraj opened the pull request.

Limraj committed:
#2116 XSS Vulnerabilities in 2.8 - fix dataSourceList.jsp, dataPointEdit.jsp, dataPointDetails.jsp, users.jsp, editMeta.jsp, eventTextRenderer.jsp, pointName.jsp, pointProperties.jsp, textRenderer.jsp; added function: removeScriptTag in common.js; added method: DataPointEditDwr.removeScriptTag;

Limraj committed:
#2116 XSS Vulnerabilities in 2.8 - revert DataPointEditDwr.java, common.js; corrected: attractor point list - editVirtual.jsp, eventTextRenderer.jsp, textRenderer.jsp, dataSourceEdit.jsp, watchList.jsp

Limraj committed:
#2116 XSS Vulnerabilities in 2.8 - refactoring: added function convertToText to common.js

Limraj committed:
#2116 XSS Vulnerabilities in 2.8 - corrected

Limraj committed:
Merge branch 'release/2.7.6' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
#2116 XSS Vulnerabilities in 2.8 - added ScadaEscapeUtils.java; fixed ChartComparatorComponent.java

Limraj committed:
Merge branch 'release/2.7.7' into fix/#2116_XSS_Vulnerabilities_in_2_8

Patrykb0802 committed:
Merge branch 'release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
Merge branch 'release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- fixed eventTextRenderer.jsp and textRenderer.jsp;
- fixed Functions.getHtml - escaped value;
- if value is null then XssProtectHtmlEscapeUtils.escape is string empty;

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- removed unused class: ScadaEscapeUtils.java;
- corrected compoundComponent/ChartComparatorComponent, displayView.tag, enhancedImageChartSeriesConfig.tag;

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- Fixed: Link, Button (script);
- In Meta data source, Script context of DataPoint is escaped;
- In Scripts, Context points is escaped;

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- Fixed: Link, Button (script);
- In Meta data source, Script context of DataPoint is escaped;
- In Scripts, Context points is escaped;
- Fixed variable name in script and Meta Data Point;

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Limraj committed:
#2669 Fixed XSS vulnerabilities in graphical view components:
- Fixed Simple Point: Name, Point name override, Style attribute, Display controls, Background color;
- Fixed Simple compound: Name, Background colour, Lead point, Sub point X;

Patrykb0802 reviewed (commented):
- In Graphical Views, Component Link is not escaped ---> it's ok now

![image](https://github.com/user-attachments/assets/865c91f9-2624-49b2-957b-0d07743ff374)

 - In Graphical Views, Component Button (script) is not escape ----> it's ok now

![image](https://github.com/user-attachments/assets/adb7dc84-3afe-433f-9a23-dd01412f6728)

 - In Meta data source, Script context of DataPoint is not escaped --> when adding data point to context in meta data point with name containing xss, alert pops up:

![image](https://github.com/user-attachments/assets/7534b9fd-dd1b-4ee6-a33d-f52fc1eebf50)

 - In Scripts, Context points is not escaped ----> ok, data point name in context is escaped:

![image](https://github.com/user-attachments/assets/b1594315-4af5-404b-9b96-136560d84f93)

Limraj committed:
#2669 Fixed XSS vulnerabilities in graphical view components:
- Fixed: <span>value</span>;
- Fixed: Background color, Style attribute;
- Fixed value alphanumeric type: added properties: datapoint.type.alphanumeric.escaped default false;

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- Fixed point context in Meta Data Point;
- Fixed reset message error after saved success;

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- In Data source edit, Name of Point is escaped;

Limraj committed:
Merge pull request #3042 from SCADA-LTS/fix/#2669_Fixed_XSS_vulnerabilities_in_graphical_view_components

#2669 Fixed XSS vulnerabilities in graphical view components:

Limraj committed:
#2116 XSS Vulnerabilities in 2.8:
- Fixed reports;
- Added dwr converters: XssDataPointBeanConverter, XssDataPointVoConverter;
- Refactor points context in Meta Data Sources, Scripts, Reports, created functions: ScriptPointsContext, ReportPointsContext

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#2116_XSS_Vulnerabilities_in_2_8

Patrykb0802 reviewed (approved):
In Meta data source, Script context of DataPoint is not escaped --> it's ok now:
![image](https://github.com/user-attachments/assets/f8b6d669-d553-403a-9745-9b553d2fbe2a)

 In Reports, Context points is not escaped --> context points are escaped correctly
![image](https://github.com/user-attachments/assets/b7ed6d86-a5d8-4c13-a2ec-44616334ddcd)
